### PR TITLE
Fb ask favor finished

### PR DESCRIPTION
**Pull request by Jmion**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sweng/favours/database/Favor.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package ch.epfl.sweng.favours.database;
+
+import android.util.Log;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Favor extends DatabaseHandler {
+
+    private static final String TAG = "c";
```
Comment (alexander-san) [inline app/src/main/java/ch/epfl/sweng/favours/database/Favor.java]: this should be corrected in the next sprint

File: app/src/main/java/ch/epfl/sweng/favours/FavoursFragment.java [hunk 2]
```diff
@@ -52,20 +53,23 @@ public void afterTextChanged(Editable editable) {
...
             public void onClick(View v) {
-                if (title.get().isEmpty()) {
-
-                    CharSequence text = "You cannot create favor with an empty title!";
-                    int duration = Toast.LENGTH_SHORT;
-                } else if (description.get().isEmpty()) {
-                    //Error cannot create favor with empty description
+                if (newFavor.get(Favor.StringFields.title).isEmpty() || newFavor.get(Favor.StringFields.description).isEmpty()) {
+                    Log.d(TAG, "Failure to check favor input");
+                    launchToast("There is an empty field left");
```
Comment (alexander-san) [inline app/src/main/java/ch/epfl/sweng/favours/FavoursFragment.java]: I'll update in the next sprint to check individually for all fields and provide descriptive toasts to the user

File: app/src/main/java/ch/epfl/sweng/favours/database/Favor.java [hunk 3]
```diff
@@ -33,6 +36,7 @@ public void updateOnDb(){
                         documentID = docRef.getId();
                         updateFromDb();
                     }).addOnFailureListener(e -> {
+                        Log.d(TAG,"failure to push to database");
                 /* Feedback of an error here - Impossible to update user informations */
```
Comment (alexander-san) [inline app/src/main/java/ch/epfl/sweng/favours/database/Favor.java]: launchToast("It is currently impossible to add the Favor. please check your connection");

File: app/src/main/java/ch/epfl/sweng/favours/FavoursFragment.java [hunk 4]
```diff
@@ -1,18 +1,51 @@
...
+
+        binding.addFavor.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (title.get().isEmpty()) {
+
+                    CharSequence text = "You cannot create favor with an empty title!";
+                    int duration = Toast.LENGTH_SHORT;
+                } else if (description.get().isEmpty()) {
+                    //Error cannot create favor with empty description
```
Comment (alexander-san) [inline app/src/main/java/ch/epfl/sweng/favours/FavoursFragment.java]: next sprint will contain a error handling and documentation issue

Comment (alexander-san) [reply app/src/main/java/ch/epfl/sweng/favours/FavoursFragment.java]: There is a lot to be done to have a fail-save user interaction that also gives descriptive output on the console in debug mode

**Comment by Jmion** [comment]
Merging but issues will be fixed by @alexander-san in his refactor branch.

**Pull request merged**